packages/dataviews/src/dataviews.tsx

```diff
@@ -79,26 +80,22 @@ export default function DataViews< Item extends AnyItem >( {
 	onSelectionChange = defaultOnSelectionChange,
 }: DataViewsProps< Item > ) {
 	const [ selectionState, setSelectionState ] = useState< string[] >( [] );
-	let selection, setSelection;
-	if (
-		selectionProperty !== undefined &&
-		setSelectionProperty !== undefined
-	) {
-		selection = selectionProperty;
-		setSelection = setSelectionProperty;
-	} else {
-		selection = selectionState;
-		setSelection = setSelectionState;
-	}
+	const selection =
```

Comment:
This is not quite the same logic as before: the `selection` could end up being `selectionState` while `setSelection` being `setSelectionProperty`.